sha2_256.c

```diff
@@ -143,6 +143,13 @@ void sha2_256_compress(void *v)
 /* initialize */
 
 static void sha2_256_init_h0(sha2_256_t *sha, const uint8_t *h0)
+__contract__(
+  requires(memory_no_alias(sha, sizeof(sha2_256_t)))
+  requires(memory_no_alias(h0, 32))
+  assigns(memory_slice(sha->s, sizeof(uint32_t)*(8+24)))
```

Comment:
We should avoid such magic numbers. This should also work:
```suggestion
  assigns(memory_slice(sha->s, sizeof(sha->s))
```

Reply:
I think it's fine for now.